src/metabase/util.clj

```diff
@@ -66,7 +65,7 @@
               (->Timestamp (.getTime this)))
   ;; Strings are expected to be in ISO-8601 format. `YYYY-MM-DD` strings *are* valid ISO-8601 dates.
   String    (->Timestamp [this]
-              (->Timestamp (DatatypeConverter/parseDateTime this)))
+              (->Timestamp (coerce/from-string this)))
```

Comment:
This is really inefficient as there's no order to which date format pasers it tries first. I wrote [this](https://github.com/metabase/metabase/blob/master/src/metabase/util.clj#L889) to help fix that.

Reply:
Why don't we just make str->datetime the implementation of ->Timestamp for strings?

There's no point in having two functions that do the same thing

Reply:
->Timestamp returns a java.sql.Timestamp (i.e. (.getMillis date-time)) and doesn't know anything about timezones. I think (->Timestamp) could call it, but I think there's lots of times we don't want a Timestamp.

Reply:
What if the strings contain a timezone :D